apisix/plugin.lua

```diff
@@ -805,4 +810,32 @@ function _M.run_global_rules(api_ctx, global_rules, phase_name)
 end
 
 
+function _M.rerun_plugins_of_consumer(plugins, api_ctx)
+    for i = 1, #plugins, 2 do
+        -- no need to rerun the auth plugins
+        if plugins[i + 1]["_from_consumer"] and plugins[i].type ~= "auth" then
```

Comment:
Would be better to merge it in https://github.com/apache/apisix/blob/492f782d9fae719b3a20e0d65c275962bcdcaab9/apisix/plugin.lua#L736?

Reply:
I had thought about it, but `if plugins[i + 1]["_from_consumer"] and plugins[i].type ~= "auth" then` is difficult to port. Unless we add a flag or something like that. This makes run_plugin look even more complicated.

Another way is as follows:

```
    for i = 1, #plugins, 2 do
        -- no need to rerun the non-auth plugins
        if plugins[i + 1]["_from_consumer"] and plugins[i].type ~= "auth" then
            local new_plugins = core.table.new(2, 0)
            core.table.insert(new_plugins, plugins[i])
            core.table.insert(new_plugins, plugins[i + 1])
            _M.run_plugin("rewrite", new_plugins, api_ctx)
        end
    end
    return api_ctx
```

this adds some table overhead.

Reply:
The main reason is that the `plugins` here are used in the `api_ctx` in order to reduce the table overhead, so use the judgment condition.

Reply:
Maybe we can add another fake phase (rewrite_in_consumer) in `run_plugin` so we don't need to duplicate the conditions?

Reply:
update